### PR TITLE
Add post-publication comment for solaris-heppa-4-3

### LINKED ISSUE #66: Solar data tiny tweak - targeting SOLARIS-HEPPA-CMIP-4-4
@st-bender @berndfunke 1000 apologies for this, I should have caught it earlier. For your next version (whenever that is), there is one tiny other tweak to your solar data. Can you please remove the attributes `datetime_end` and `datetime_start` global attributes in all your files. They're not written correctly at the moment (the format is meant to depend on the file's frequency, a bit of a headache) and we can actually just infer them from the time axis in your data, so we can avoid sparing you the details of the frequency-dependent formatting.

Apologies again, I know we asked you to put that there in the first place. For the 4.3 data, it doesn't matter, just leave it as is.

One more thing for the list: we should double check the creation_date attribute. This should be picked up by the new version of input4mips-validation, but just in case, here is a reminder. If the solar team want to write this themselves, it must be the timestamp when the file was created (in the UTC timezone), in the format "YYYY-mm-ddTHH:MM:SSZ" (the ISO8601 standard).

Thanks and sorry again!

cc @durack1

### PR DESCRIPTION
**Pull request by znichollscr**
## Description

Closes #66 

## Checklist

Please confirm that this pull request has done the following:

- [x] Documentation added (where applicable)
- [x] Changelog item added to `changelog/`

### CODE REVIEW

**Comment by znichollscr** [comment]
@durack1 FYI, getting ready for moving to SOLARIS-HEPPA 4.4

File: Database/input-data/pmount/031e922a53f27d2c3f48656fef07c655c7418103de14fbac4d21b083d1ece50c.json [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    "grid_label":"gn",
+    "institution":"APARC SOLARIS-HEPPA",
+    "institution_id":"SOLARIS-HEPPA",
+    "latest":null,
+    "license":"Solar forcing data produced by SOLARIS-HEPPA is licensed under a Creative Commons Attribution 4.0 International License (http://creativecommons.org/licenses/by/4.0/). The data producers and data providers make no warranty, either expressed or implied, including but not limited to, warranties of merchantability and fitness for a particular purpose. All liabilities arising from the supply of the information (including any liability arising in negligence) are excluded to the fullest extent permitted by law.",
+    "license_id":null,
+    "mip_era":"CMIP6Plus",
+    "nominal_resolution":"10000 km",
+    "product":"derived",
+    "publication_status":"in_publishing_queue",
```
Comment (durack1) [inline Database/input-data/pmount/031e922a53f27d2c3f48656fef07c655c7418103de14fbac4d21b083d1ece50c.json:26]: This looks wrong - where is this entry found?

File: docs/database-views/input4MIPs_delivery-summary_CMIP6Plus.html [hunk 2]
```diff
@@ -100,9 +100,9 @@ <h4>
     <tr>
       <td>8</td>
       <td><a href='http://solarisheppa.geomar.de/cmip7' target='_blank'>Solar</a></td>
-      <td>SOLARIS-HEPPA-CMIP-4-3</td>
+      <td>SOLARIS-HEPPA-CMIP-4-4</td>
       <td>Preliminary dataset available</td>
-      <td><a href='https://aims2.llnl.gov/search?project=input4MIPs&versionType=all&activeFacets=%7B%22source_id%22%3A%22SOLARIS-HEPPA-CMIP-4-3%22%7D' target='_blank'>Available: v4.3 (1850-01-01 to 2023-12-31)</a></td>
```
Comment (durack1) [inline docs/database-views/input4MIPs_delivery-summary_CMIP6Plus.html:105]: This also looks wrong

File: docs/database-views/input4MIPs_source-id_CMIP6Plus.html [hunk 3]
```diff
@@ -141,6 +141,21 @@ <h4>
     </tr>
     <tr>
       <td>5</td>
+      <td><a href='https://github.com/PCMDI/input4MIPs_CVs/blob/main/CVs/input4MIPs_source_id.json' target='_blank'>In publishing queue</a></td>
+      <td>CMIP6Plus</td>
+      <td>SOLARIS-HEPPA-CMIP-4-4</td>
+      <td>None</td>
+      <td>in_publishing_queue</td>
```
Comment (durack1) [inline docs/database-views/input4MIPs_source-id_CMIP6Plus.html:148]: Also wrong

File: docs/database-views/input4MIPs_source-id_CMIP6Plus.html [hunk 4]
```diff
@@ -200,13 +215,13 @@ <h4>
       <td>1.1.9</td>
     </tr>
     <tr>
-      <td>9</td>
+      <td>10</td>
       <td><a href='https://aims2.llnl.gov/search?project=input4MIPs&versionType=all&activeFacets=%7B%22source_id%22%3A%22SOLARIS-HEPPA-CMIP-4-3%22%2C%22mip_era%22%3A%22CMIP6Plus%22%7D' target='_blank'>Published</a></td>
       <td>CMIP6Plus</td>
       <td>SOLARIS-HEPPA-CMIP-4-3</td>
       <td>True</td>
       <td>published</td>
```
Comment (durack1) [inline docs/database-views/input4MIPs_source-id_CMIP6Plus.html:223]: should be retracted

**Review by durack1 — COMMENTED** [review]
Something looks wrong here - not sure where incorrect information is hiding. There may be a need to update the `esgf-input4MIPs.json` I might have jumped ahead of myself a little earlier than the publication time, so I am chasing the latest info

**Comment by durack1** [comment]
Ok and trying again, I hit
```python
(xcd072nctax) (venv) (xcd072nctax) bash-4.2$ source venv/bin/activate
(venv) (xcd072nctax) bash-4.2$ python python-packages/input4MIPs-CVs/src/input4MIPs_CVs/cli/update-database.py --repo-root-dir .
Loading pmount entries: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1557/1557 [00:00<00:00, 3958.72it/s]
Adding ESGF data to pmount records: 1557it [00:00, 3806.78it/s]
Source ID entry missing from CVs:
{
    "SOLARIS-HEPPA-CMIP-4-4":{
        "contact":"bernd@iaa.es",
        "further_info_url":"http://solarisheppa.geomar.de/cmip7",
        "institution_id":"SOLARIS-HEPPA",
        "license_id":null,
        "mip_era":"CMIP6Plus",
        "source_version":"4.4"
    }
}
...
```

**Comment by durack1** [comment]
@znichollscr I have run out of steam here, and this requires a source_id registration, which is not in place to occur before the rest will make sense - if you can point out how to do that, I should be able to augment my notes for next go around

File: Database/input-data/pmount/031e922a53f27d2c3f48656fef07c655c7418103de14fbac4d21b083d1ece50c.json (hunk 1 above)
Comment (znichollscr) [reply Database/input-data/pmount/031e922a53f27d2c3f48656fef07c655c7418103de14fbac4d21b083d1ece50c.json:26]: It's just a placeholder. When we generate these entries, we don't actually know the publication status (sometimes we get the files before they're published). This gets updated as part of updating the database.

**Comment by znichollscr** [comment]
> @znichollscr I have run out of steam here

All good I got it.

> this requires a source_id registration, which is not in place to occur before the rest will make sense - if you can point out how to do that, I should be able to augment my notes for next go around

Update `CVs/input4MIPs_source_id.json`. I've done that now here.

File: Database/input4MIPs_db_file_entries.json [hunk 5]
```diff
@@ -773,6 +773,49 @@
...
+        "nominal_resolution":"10000 km",
+        "product":"derived",
+        "publication_status":"in_publishing_queue",
+        "realm":"atmos",
+        "references":"Funke et al., Geosci. Model Dev., 17, 1217--1227, https://doi.org/10.5194/gmd-17-1217-2024, 2024",
+        "region":"global",
+        "replica":null,
+        "sha256":"031e922a53f27d2c3f48656fef07c655c7418103de14fbac4d21b083d1ece50c",
+        "source":"SSI, TSI, and F10.7 from ssi_v03r00 (Odele Coddington et al., pers. comm.); Ap and Kp from ftp.ngdc.noaa.gov until 2014, afterwards from GFZ Potsdam (https://kp.gfz-potsdam.de), P-IPR from SEP-II (Ilya Usoskin et al., pers. comm.), MEE-IPR from FMI APEEP v2024b_cmip7 (Max van de Kamp et al., pers. comm.), GCR-IPR from CRII v2024-02 (Ilya Usoskin et al., pers. comm.)",
+        "source_id":"SOLARIS-HEPPA-CMIP-4-4",
```
Comment (znichollscr) [inline Database/input4MIPs_db_file_entries.json:808]: @durack1 just FYI if you want to check the raw JSON this is where you should look. It's pretty hard to parse though so I rely on the HTML pages mostly.

File: docs/database-views/input4MIPs_delivery-summary_CMIP6Plus.html (hunk 2 above)
Comment (znichollscr) [reply docs/database-views/input4MIPs_delivery-summary_CMIP6Plus.html:105]: Updated now.

File: docs/database-views/input4MIPs_source-id_CMIP6Plus.html (hunk 4 above)
Comment (znichollscr) [reply docs/database-views/input4MIPs_source-id_CMIP6Plus.html:208]: This is now 4.4, 4.3 is retracted below.

**Comment by znichollscr** [comment]
Alrighty @durack1 good to go

File: Database/input-data/pmount/031e922a53f27d2c3f48656fef07c655c7418103de14fbac4d21b083d1ece50c.json (hunk 1 above)
Comment (durack1) [inline Database/input-data/pmount/031e922a53f27d2c3f48656fef07c655c7418103de14fbac4d21b083d1ece50c.json:26]: This is published now - so this needs a toggle

File: Database/input-data/pmount/044136f5ef82813ecd62b401ab708425d6cdd5cc5151d57d95bb8db8325ce5d3.json [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    "grid_label":"gn",
+    "institution":"APARC SOLARIS-HEPPA",
+    "institution_id":"SOLARIS-HEPPA",
+    "latest":null,
+    "license":"Solar forcing data produced by SOLARIS-HEPPA is licensed under a Creative Commons Attribution 4.0 International License (http://creativecommons.org/licenses/by/4.0/). The data producers and data providers make no warranty, either expressed or implied, including but not limited to, warranties of merchantability and fitness for a particular purpose. All liabilities arising from the supply of the information (including any liability arising in negligence) are excluded to the fullest extent permitted by law.",
+    "license_id":null,
+    "mip_era":"CMIP6Plus",
+    "nominal_resolution":"10000 km",
+    "product":"derived",
+    "publication_status":"in_publishing_queue",
```
Comment (durack1) [inline Database/input-data/pmount/044136f5ef82813ecd62b401ab708425d6cdd5cc5151d57d95bb8db8325ce5d3.json:26]: Ditto

File: Database/input-data/pmount/f0af84602b1a7ec3c884fbfe8d85b10013528ce508c6d4302992ebdaa5545c13.json [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    "grid_label":"gn",
+    "institution":"APARC SOLARIS-HEPPA",
+    "institution_id":"SOLARIS-HEPPA",
+    "latest":null,
+    "license":"Solar forcing data produced by SOLARIS-HEPPA is licensed under a Creative Commons Attribution 4.0 International License (http://creativecommons.org/licenses/by/4.0/). The data producers and data providers make no warranty, either expressed or implied, including but not limited to, warranties of merchantability and fitness for a particular purpose. All liabilities arising from the supply of the information (including any liability arising in negligence) are excluded to the fullest extent permitted by law.",
+    "license_id":null,
+    "mip_era":"CMIP6Plus",
+    "nominal_resolution":"10000 km",
+    "product":"derived",
+    "publication_status":"in_publishing_queue",
```
Comment (durack1) [inline Database/input-data/pmount/f0af84602b1a7ec3c884fbfe8d85b10013528ce508c6d4302992ebdaa5545c13.json:26]: ditto

**Review by durack1 — APPROVED** [review]
@znichollscr looks good to me, other than that remaining in publishing queues - but I suppose these are the mount hash files, so does it matter?

File: Database/input-data/pmount/031e922a53f27d2c3f48656fef07c655c7418103de14fbac4d21b083d1ece50c.json (hunk 1 above)
Comment (znichollscr) [reply Database/input-data/pmount/031e922a53f27d2c3f48656fef07c655c7418103de14fbac4d21b083d1ece50c.json:26]: I think the previous comment got lost. This is just a placeholder. These files are generated from the stuff from pmount, which doesn't know anything about the status on ESGF, hence this value will always be wrong. If you want to look at the JSON (which I don't recommend, because it's painful to parse), you have to look at this file: Database/input4MIPs_db_file_entries.json

File: Database/input-data/pmount/f0af84602b1a7ec3c884fbfe8d85b10013528ce508c6d4302992ebdaa5545c13.json (hunk 7 above)
Comment (znichollscr) [reply Database/input-data/pmount/f0af84602b1a7ec3c884fbfe8d85b10013528ce508c6d4302992ebdaa5545c13.json:26]: See https://github.com/PCMDI/input4MIPs_CVs/pull/137#discussion_r1810676264

**Comment by znichollscr** [comment]
> but I suppose these are the mount hash files, so does it matter?

Exactly (these files don't know about ESGF state, so the field you're talking about has to be wrong). For further details, see https://github.com/PCMDI/input4MIPs_CVs/pull/137#discussion_r1810676264

**Comment by durack1** [comment]
Ok, merge away!
